models/challenges.js

```diff
@@ -66,10 +94,10 @@ const postChallenge = async (challengeData) => {
 const subscribeUserToChallenge = async (userId, challengeId) => {
   try {
     const getUser = await userModel.doc(userId).get()
-    const user = getUser.data().github_display_name
+    const user = getUser.data()
     if (user) {
       const challengeRef = await challengesModel.doc(challengeId)
-      await challengeRef.update({ participants: Firestore.FieldValue.arrayUnion({ name: user }) })
+      await challengeRef.update({ participants: Firestore.FieldValue.arrayUnion({ id: userId }) })
```

Comment:
What are you trying to do here?

Reply:
as participants should store userIds and not username, so replacing user with id.

Reply:
How is this getting stored?
Objects inside the array?